src/pos/tasks/ot_croissant.py

```diff
@@ -66,7 +67,14 @@ def run(self) -> Self:
 
         # Converting the list of paths to a list of strings and prepending the work_path
         logger.debug('converting the list of paths to a list of strings and prepending the work_path')
-        datasets = [str(self.context.config.work_path / p) for p in self.spec.dataset_paths]
+        # Get the directory path from self.spec.dataset_paths
+        directory_path = str(self.context.config.work_path / self.spec.dataset_path)
+
+        # List all sub-folders in the directory
+        datasets = [
+            str(self.context.config.work_path / self.spec.dataset_path / dataset)
+            for dataset in os.listdir(directory_path) if os.path.isdir(os.path.join(directory_path, dataset))
+        ]
```

Comment:
```suggestion
        datasets = [str(path) for path in Path(directory_path).iterdir() if path.is_dir()]
```

You can remove the `os` import if you do it this way.

Reply:
I was suspecting Path might be able to do that... I should never be lazy again. :D